### PR TITLE
feat: Rename `wasmer-engine-object-file` to `wasmer-engine-staticlib`.

### PR DESCRIPTION
**Pull request by Hywan**
# Description

This PR renames the `wasmer-engine-object-file` crate into `wasmer-engine-staticlib`. We believe it conveys a better meaning of what it does. This PR is similar to its siblings, #2340 and #2356.

### CODE REVIEW

**Comment by Hywan** [comment]
bors try

File: lib/engine-staticlib/Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "wasmer-engine-object-file"
+name = "wasmer-engine-staticlib"
 version = "1.0.2"
 authors = ["Wasmer Engineering Team <engineering@wasmer.io>"]
 description = "Wasmer Object File Engine"
```
Comment (jubianchi) [inline lib/engine-staticlib/Cargo.toml:5]: Should we change the description here?

**Review by jubianchi — APPROVED** [review]

File: lib/engine-staticlib/Cargo.toml (hunk 1 above)
Comment (Hywan) [reply lib/engine-staticlib/Cargo.toml:5]: We should indeed! Thanks!

**Pull request merged**